src/isomorphic/classic/element/ReactElement.js

```diff
@@ -284,14 +299,20 @@ ReactElement.cloneElement = function(element, config, children) {
         'Properties defined in its prototype chain will be ignored.'
       );
     }
-    if (config.ref !== undefined) {
+
+    var configRef = getRef(config);
+    var configKey = getKey(config);
+
+    if (configRef != null) {
       // Silently steal the ref from the parent.
```

Comment:
If `null` values should be allowed here, I think we'd need to check `config.ref` and `config.key` before calling `getRef` and `getKey` and then always set `ref` and `key` to the return.